src/concrete/ml/sklearn/tree_to_numpy.py

```diff
@@ -370,7 +371,7 @@ def tree_to_numpy(
     )
 
     # Execute with 1 example for efficiency in large data scenarios to prevent slowdown
-    onnx_model = get_onnx_model(model, x[:1], framework)
+    onnx_model = get_onnx_model(model, x[:2] if x.shape[0] > 1 else x, framework)
```

Comment:
worth some comments here so that we remember why this was done in the future

Reply:
I added comments and FIXME

Reply:
thanks